mkosi/__init__.py

```diff
@@ -3260,6 +3260,26 @@ def run_finalize_script(args: MkosiArgs, root: Path, do_run_build_script: bool,
         run([args.finalize_script, verb], env=env)
 
 
+def make_reproducible(args: MkosiArgs, root: Path, do_run_build_script: bool, for_cache: bool) -> None:
+    verb = "build" if do_run_build_script else "final"
+
+    with complete_step("Attempting to make build reproducible…"):
+        env = collections.ChainMap(dict(BUILDROOT=str(root), OUTPUTDIR=str(output_dir(args))),
+                                   args.environment,
+                                   os.environ)
+
+        file = f"reproducible.{args.distribution}.sh"
+        if importlib.resources.is_resource("mkosi.resources", file):
+            with importlib.resources.path("mkosi.resources", file) as p:
+                run(["bash", p, verb], env=env)
```

Comment:
More in line with the rest of the functions would be to match `args.distribution` and call some `make_reproducible_$DISTRO`. This could be something like
```python
def make_reproducible_debian(args, root):
    for p in ("var/log/dpkg.log", "var/log/bootstrap.log", "var/cache/apt/pkgcache.bin", "var/log/apt/history.log", "var/log/apt/term.log", "var/log/alternatives.log", "var/cache/ldconfig/aux-cache", "var/log/apt/eipp.log.xz", "var/lib/dbus/machine-id"):
        (root / p).unlink(missing_ok=True)
```
(with the caveat, that we are still on Python 3.7 so `missing_ok=` doesn't yet exist, so a try-except block would be needed).

In the case of keeping the extra script: `run` passes keyword arguments it doesn't use through, so instead calling bash yourself you could pass in `shell=True`.

Reply:
We could do something like that. But currently I think the bash stuff is easier to dry-run and experiment with. But I might be alone with that opinion :)